### PR TITLE
Add DA.Set to match DA.Map

### LINKED ISSUE #8448: Bring `DA.Map` and `DA.Set` to feature parity with `DA.Next.*`

### PR DESCRIPTION
**Pull request by sofiafaro-da**
Fixes #8448

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/daml-stdlib-src/DA/Set.daml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  , union
+  , intersection
+  , difference
+  ) where
+
+import DA.Map (Map)
+import DA.Map qualified as M
+
+-- | The type of a set. This is a wrapper over the `Map` type.
+data Set k = Set with map : Map k ()
```
Comment (cocreature) [inline compiler/damlc/daml-stdlib-src/DA/Set.daml:66]: We should move this to a stable type. iirc the only reason why we didn’t do it for `DA.Next.Set` was that it was soon going to be deprecated which turned out to be not quite as soon :grin: Doesn’t have to be in this PR

**Review by cocreature — APPROVED** [review]
Awesome thanks! It would be great to copy https://github.com/digital-asset/daml/blob/main/compiler/damlc/tests/daml-test-files/Set.daml to a version based on this.

File: compiler/damlc/daml-stdlib-src/DA/Set.daml (hunk 1 above)
Comment (sofiafaro-da) [reply compiler/damlc/daml-stdlib-src/DA/Set.daml:66]: Good point!

**Comment by sofiafaro-da** [comment]
I moved Set to a stable package and copied the old DA.Next.Set test, turning it into a DA.Set test, but keeping the old test around for now.

**Pull request merged**